fix: remove `turbo` column

wa0x6e opened the pull request:
Remove the legacy `turbo` boolean column, using `turbo_expiration` as the sole source of truth for a space's turbo status.

- **Schema/fixtures**: Drop `turbo` from `schema.sql`, test fixtures
- **`src/helpers/spaces.ts`**: Remove `turbo` from SELECT queries; compute turbo status via `isTurbo(turbo_expiration)` comparing against current timestamp
- **`src/helpers/metrics.ts`**: Metrics now query `turbo_expiration > UNIX_TIMESTAMP()` for active turbo count (single UNION ALL query)
- **GraphQL**: Remove `turbo` from space mappings, queries, and schema references across operations
- **Tests**: Drop `turbo` field assertions, add `turbo_expiration`-based turbo space test case

```ts
function isTurbo(turboExpiration: number): boolean {
  return turboExpiration > Date.now() / 1e3;
}
```

wa0x6e committed:
fix: remove `turbo` column

ChaituVR commented:
We updated `turbo_expiration` for all turbo spaces already?

wa0x6e commented:
> We updated `turbo_expiration` for all turbo spaces already?

Done for most of them, still a few remaining.

This PR should be merge only once @bonustrack confirmed that it's done, then if everything working, we could delete the `turbo` column in the database in a second time

bonustrack commented:
Not all the space are updated, we still miss the spaces who got free turbo because of their network plan. Will update here once its done

ChaituVR commented:
@copilot+claude-sonnet-4.6 Fix conflicts

Copilot committed:
chore: merge master into fix-remove-turbo-column, resolve conflicts

Co-authored-by: ChaituVR <15967809+ChaituVR@users.noreply.github.com>

Copilot committed:
chore: merge master into fix-remove-turbo-column, resolve conflicts

Agent-Logs-Url: https://github.com/snapshot-labs/snapshot-hub/sessions/56ed8409-9c68-4b9e-bf56-f416d5613bae

Co-authored-by: ChaituVR <15967809+ChaituVR@users.noreply.github.com>

ChaituVR commented:
@copilot+claude-opus-4.6 Review all changes, i think we won't need changes in package-lock and yarn.lock, remove changes from those

Copilot committed:
chore: remove package-lock.json and revert yarn.lock changes

Agent-Logs-Url: https://github.com/snapshot-labs/snapshot-hub/sessions/15507322-079d-4eb0-814b-00753fcaf7ef

Co-authored-by: ChaituVR <15967809+ChaituVR@users.noreply.github.com>

bonustrack commented:
We still have a ton of space with turbo=1 and no expiration, we need to figure out which ones should not be on Turbo anymore, which one should be and what are the expiration. Maybe @aurelianoB can help here. These are the spaces with turbo and no expiration:

beets.eth
bitdao.eth
councilofrivendell.eth
cow.eth
defi.app
gnosis.eth
info.sonic
rbnt.eth
sd-yieldbasis.eth
sdapw.eth
sdbal.eth
sdbpt.eth
sdcake.eth
sdcrv-gov.eth
sdcrv.eth
sdfxn.eth
sdfxs.eth
sdpendle.eth
sdspectra.eth
sdyfi.eth
sdynd.eth
sdzero.eth
signaling.1inch.eth
stakedao.eth
starknet.eth
starknetdemo.eth
superfluid.eth
superraredao.eth
telcoinappvote.eth
telcoincompliancecouncil.eth
telcoinplatformcouncil.eth
telcoinstakervote.eth
telcointancouncil.eth
telcointreasurycouncil.eth
telcoinvalidatorvote.eth
telxcouncil.eth
telxvote.eth
umerge.eth
xmaquina.eth

aurelianoB commented:
Here's the list with expiration date associated

beets.eth: May 5, 2026
bitdao.eth: Feb 6, 2027
councilofrivendell.eth: Aug 12, 2026
cow.eth: Oct 29, 2026
defi.app: Apr 15, 2026
gnosis.eth: Mar 18, 2027
info.sonic: Jan 22, 2027
rbnt.eth: Aug 19, 2026
sd-yieldbasis.eth: Feb 13, 2027
sdapw.eth: Feb 13, 2027
sdbal.eth: Feb 13, 2027
sdbpt.eth: Feb 13, 2027
sdcake.eth: Feb 13, 2027
sdcrv-gov.eth: Feb 13, 2027
sdcrv.eth: Feb 13, 2027
sdfxn.eth: Feb 13, 2027
sdfxs.eth: Feb 13, 2027
sdpendle.eth: Feb 13, 2027
sdspectra.eth: Feb 13, 2027
sdyfi.eth: Feb 13, 2027
sdynd.eth: Feb 13, 2027
sdzero.eth: Feb 13, 2027
signaling.1inch.eth: Sep 22, 2026
stakedao.eth: Feb 13, 2027
starknet.eth: Apr 15, 2027
starknetdemo.eth: Apr 15, 2027
superfluid.eth: Apr 15, 2026
superraredao.eth: Sep 17, 2026
telcoinappvote.eth: Aug 12, 2026
telcoincompliancecouncil.eth: Aug 12, 2026
telcoinplatformcouncil.eth: Aug 12, 2026
telcoinstakervote.eth: Aug 12, 2026
telcointancouncil.eth: Aug 12, 2026
telcointreasurycouncil.eth: Aug 12, 2026
telcoinvalidatorvote.eth: Aug 12, 2026
telxcouncil.eth: Aug 12, 2026
telxvote.eth: Aug 12, 2026
umerge.eth: Sep 30, 2026
xmaquina.eth: Apr 15, 2026

bonustrack commented:
Updated all the spaces, we should be able to remove turbo column now

wa0x6e commented:
Once merged, we should confirm that all the turbo spaces are correct, then drop the `turbo` column from the database

bonustrack commented:
Shouldn't we remove Turbo param from the UI first?

wa0x6e commented:
> Shouldn't we remove Turbo param from the UI first?

No, `turbo` is still the valid filter in graphql, we're removing only the `turbo` column in the database

bonustrack reviewed (approved):
utACK
